### PR TITLE
h264e: Replace GetExtBuffer on GetExtBufferRef

### PR DESCRIPTION
**Pull request by Losev-Ivan**
Signed-off-by: Ivan Losev <ivan.losev@intel.com>

### CODE REVIEW

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 1]
```diff
@@ -7516,17 +7516,17 @@ void MfxVideoParam::SyncVideoToCalculableParam()
 
     if (IsMvcProfile(mfx.CodecProfile))
     {
-        mfxExtMVCSeqDesc * extMvc = GetExtBuffer(*this);
-        if (extMvc->NumView)
+        mfxExtMVCSeqDesc & extMvc = GetExtBufferRef(*this);
```
Comment (DenWolf) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: It's not a good idea to use this method inside class MfxVideoParam - there is no 100% guarantee that needed buffer is existed, so there is a way to undefined behavior.
As result, if pointer extMvc can be nullptr, it's better to check it directly, for example in the "if" condition at line 7520

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 2]
```diff
@@ -1463,14 +1463,14 @@ mfxStatus MfxHwH264Encode::ReadSpsPpsHeaders(MfxVideoParam & par)
         if (extBits.SPSBuffer)
         {
             InputBitstream reader(extBits.SPSBuffer, extBits.SPSBufSize);
-            mfxExtSpsHeader * extSps = GetExtBuffer(par);
-            ReadSpsHeader(reader, *extSps);
+            mfxExtSpsHeader & extSps = GetExtBufferRef(par);
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: ok

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 3]
```diff
@@ -1463,14 +1463,14 @@ mfxStatus MfxHwH264Encode::ReadSpsPpsHeaders(MfxVideoParam & par)
...
+            ReadSpsHeader(reader, extSps);
 
             if (extBits.PPSBuffer)
             {
                 InputBitstream pps_reader(extBits.PPSBuffer, extBits.PPSBufSize);
-                mfxExtPpsHeader * extPps = GetExtBuffer(par);
-                ReadPpsHeader(pps_reader, *extSps, *extPps);
+                mfxExtPpsHeader & extPps = GetExtBufferRef(par);
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: ok

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 1 above)
Comment (mgonchar) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: Here starts the list of buffers which can be safely extracted with GetExtBufferRef: https://github.com/Intel-Media-SDK/MediaSDK/blob/dd25daaa1ff50d8ee41f4c14b8599db3e11d2331/_studio/mfx_lib/shared/include/mfx_h264_enc_common_hw.h#L547

mfxExtMVCSeqDesc is not in this list so you can't use GetExtBufferRef for it

**Pull request merged**